src/Annotation.php

```diff
@@ -63,17 +73,85 @@ public function getLabelAndUsers(array $categories): array
 
     public function getPoints(): array
     {
-        if ($this->getShape()->id === Shape::circleId()) {
-            return $this->getCirclePoints();
+        if ($this->segmentation !== null) {
+            // Handle points first (2 coordinates)
+            if (count($this->segmentation) === 2) {
+                return $this->segmentation;
+            }
+            
+            // Handle circles specially - check if it's a circle based on raw data
+            if ($this->isCircleShapeFromSegmentation()) {
+                return $this->getCirclePoints();
+            }
+            
+            return $this->segmentation;
```

Comment:
Why change the behavior of detecting the shape here? Was there an issue with the previous implementation?